client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -282,15 +283,36 @@ public void performRequestAsync(String method, String endpoint, Map<String, Stri
     public void performRequestAsync(String method, String endpoint, Map<String, String> params,
                                     HttpEntity entity, HttpAsyncResponseConsumerFactory httpAsyncResponseConsumerFactory,
                                     ResponseListener responseListener, Header... headers) {
-        URI uri = buildUri(pathPrefix, endpoint, params);
+        Objects.requireNonNull(params, "params must not be null");
+        Map<String, String> requestParams = new HashMap<>(params);
+        Set<Integer> ignoreErrorCodes = new HashSet<>();
```

Comment:
should we initialize this conditionally and if `ignoreString == null` we just use `Collections.emptySet()`?